docs/wcs/wcsapi.rst

```diff
@@ -22,7 +22,128 @@ in Python. This API is described in the Astropy Proposal for Enhancements (APE)
 
 The core astropy package provides base classes that define the low- and high-
 level APIs described in APE 14 in the :mod:`astropy.wcs.wcsapi` module, and
-these are listed below.
+these are listed in the `Reference/API`_ section below.
+
+Pixel conventions
+=================
+
+This API assumes that integer pixel values fall at the center of pixels (as
+assumed in the FITS-WCS standard, see Section 2.1.4 of `Greisen et al., 2002,
+A&A 446, 747 <https://doi.org/10.1051/0004-6361:20053818>`_), while at the same
+time matching the Python 0-index philosophy.  That is, the first pixel is
+considered pixel ``0``, but pixel coordinates ``(0, 0)`` are the *center* of
+that pixel.  Hence the first pixel spans pixel values ``-0.5`` to ``0.5``.
+
+The order of the pixel coordinates (``(x, y)`` vs ``(row, column)``) depends on
+the method or property used, and this can normally be determined from the
+property or method name. Properties and methods containing ``pixel`` assume
+``(x, y)`` ordering, while properties and methods containing ``array`` assume
+``(row, column)`` ordering.
```

Comment:
Would it help to say:
```
Properties and methods containing ``pixel`` assume
``(x, y)`` or ``Carstesian`` ordering, while properties and methods containing ``array`` assume
``(row, column)`` or ``numpy array`` ordering.

```

Reply:
I guess that doesn't help because as our discussions revealed, I don't really get what "Cartesian ordering" means: to me "Cartesian" means "the metric is s^2 = x^2 + y^2", which is true regardless of the ordering.  And "numpy array" also doesn't help because it can be both fortran *or* C ordered.

If preferred I can write the words on it... once I actually have right in my head which is supposed to be which :wink:

Reply:
I'll push some changes soon that are similar to @nden's suggestion.